keras/ops/image.py

```diff
@@ -501,3 +501,207 @@ def map_coordinates(
         fill_mode,
         fill_value,
     )
+
+
+class PadImages(Operation):
+    def __init__(
+        self,
+        top_padding,
+        bottom_padding,
+        left_padding,
+        right_padding,
+        target_height,
+        target_width,
+    ):
+        super().__init__()
+        self.top_padding = top_padding
+        self.bottom_padding = bottom_padding
+        self.left_padding = left_padding
+        self.right_padding = right_padding
+        self.target_height = target_height
+        self.target_width = target_width
+
+    def call(self, images):
+        return _pad_images(
+            images,
+            self.top_padding,
+            self.bottom_padding,
+            self.left_padding,
+            self.right_padding,
+            self.target_height,
+            self.target_width,
+        )
+
+    def compute_output_spec(self, images):
+        if self.target_height is None:
+            height_axis = 0 if len(images.shape) == 3 else 1
+            self.target_height = (
+                self.top_padding
+                + images.shape[height_axis]
+                + self.bottom_padding
+            )
+        if self.target_width is None:
+            width_axis = 0 if len(images.shape) == 3 else 2
+            self.target_width = (
+                self.left_padding
+                + images.shape[width_axis]
+                + self.right_padding
+            )
+        out_shape = (
+            images.shape[0],
+            self.target_height,
+            self.target_width,
+            images.shape[-1],
+        )
+        if len(images.shape) == 3:
+            out_shape = out_shape[1:]
+        return KerasTensor(
+            shape=out_shape,
+            dtype=images.dtype,
+        )
+
+
+def _pad_images(
+    images,
+    top_padding,
+    bottom_padding,
+    left_padding,
+    right_padding,
+    target_height,
+    target_width,
+):
+    images = backend.convert_to_tensor(images)
+    is_batch = True
+    images_shape = images.shape
+    if len(images_shape) == 3:
+        is_batch = False
+        images = backend.numpy.expand_dims(images, 0)
+    elif len(images_shape) != 4:
+        raise ValueError(
+            f"'images' (shape {images_shape}) must have "
+            "either 3 or 4 dimensions."
+        )
+
+    batch, height, width, depth = images.shape
+
+    if [top_padding, bottom_padding, target_height].count(None) > 1:
```

Comment:
Let's switch the `>` to `!=`. This will be more general and it will match the error message perfectly.